### PR TITLE
Move Atom benchmark GPU jobs into nightly GPU job run.

### PR DESCRIPTION
**Pull request by jromnoa**
- Splitting the benchmark GPU jobs from the normal GPU jobs will help prevent downstream issues causing the tests to fail such as memory issues, hard locks, or crashes from upstream jobs.
- Tested this in a custom branch job, sample output:
```
17:52:02      Start  35: pytest_sanity_benchmark_requires_gpu.benchmark::TEST_RUN
17:52:03  1/2 Test  #35: pytest_sanity_benchmark_requires_gpu.benchmark::TEST_RUN .............   Passed    0.50 sec
17:52:03      Start 168: AutomatedTesting::Atom_TestSuite_Benchmark_GPU.benchmark::TEST_RUN
17:54:39  2/2 Test #168: AutomatedTesting::Atom_TestSuite_Benchmark_GPU.benchmark::TEST_RUN ...   Passed  143.13 sec
17:54:39  
17:54:39  100% tests passed, 0 tests failed out of 2
17:54:39  
17:54:39  Label Time Summary:
17:54:39  COMPONENT_Atom                  = 143.13 sec*proc (1 test)
17:54:39  FRAMEWORK_pytest                = 143.63 sec*proc (2 tests)
17:54:39  REQUIRES_gpu                    = 143.63 sec*proc (2 tests)
17:54:39  SUITE_benchmark                 = 143.63 sec*proc (2 tests)
17:54:39  SUITE_benchmark_REQUIRES_gpu    = 143.63 sec*proc (2 tests)
17:54:39  
17:54:39  Total Test time (real) = 143.79 sec
```

### CODE REVIEW

File: scripts/build/Platform/Windows/build_config.json [hunk 1]
```diff
@@ -175,6 +175,28 @@
...
     }
   },
+  "benchmark_test_gpu_profile": {
+    "TAGS": [
+      "nightly-incremental-internal",
+      "weekly-clean-internal",
+      "weekly-build-metrics"
+    ],
+    "PIPELINE_ENV":{
+      "NODE_LABEL":"windows-gpu-test"
```
Comment (amzn-changml) [inline scripts/build/Platform/Windows/build_config.json:185]: You'd want to change that back to `windows-gpu`

File: scripts/build/Platform/Windows/build_config.json [hunk 2]
```diff
@@ -175,6 +175,28 @@
       "TEST_SCREENSHOTS": "True"
     }
   },
+  "benchmark_test_gpu_profile": {
```
Comment (amzn-changml) [inline scripts/build/Platform/Windows/build_config.json:178]: We'd eventually want to reuse the artifacts from the initial windows profile gpu test build job to save time and resources, but we can improve on that later. See the `profile_pipe` section to see how that's done. It basically calls each job in an ordered list.

Comment (jromnoa) [reply scripts/build/Platform/Windows/build_config.json:178]: we can revisit this later

**Review by smurly — APPROVED** [review]
do we have a linux version of this benchmark or is it windows-gpu only as seen here?

**Review by amzn-changml — APPROVED** [review]

**Comment by amzn-changml** [comment]
> do we have a linux version of this benchmark or is it windows-gpu only as seen here?

We need to get the Linux GPU nodes online. They're ready to deploy just need to clear up dependencies like driver version

**Pull request merged**